### PR TITLE
Add a version.edn cache along version.clj

### PR DESCRIPTION
**Pull request by jgrodziski**
My use case is to be able to `slurp` the file `version.edn` in project.clj with a simple 
```
~(str (:version (clojure.edn/read-string (slurp "src/version.edn"))))
```
to include the version in any filename produced.

### CODE REVIEW

File: src/leiningen/v/file.clj [hunk 1]
```diff
@@ -10,6 +10,14 @@
                      (format "(def raw-version \"%s\")" describe)
                      ""]))
 
+(defn- cache-edn-source
+  "return EDN data structure for the version cache EDN file"
```
Comment (cch1) [inline src/leiningen/v/file.clj]: This should use `pr-str` instead of manually generating the edn.

File: project.clj [hunk 2]
```diff
@@ -21,6 +21,6 @@
   :license {:name "Eclipse Public License"
             :url "http://www.eclipse.org/legal/epl-v10.html"}
   :jvm-opts ["-Djava.io.tmpdir=./tmp"]
-  :repositories [["rk-public" {:url "http://rk-maven-public.s3-website-us-east-1.amazonaws.com/releases/"}]
```
Comment (cch1) [inline project.clj]: These  changes should be part of a separate pull request.

File: test/integration/leiningen/v.clj [hunk 3]
```diff
@@ -22,7 +22,7 @@
   (against-background (before :facts (do (clone! "test0.repo") (commit!))))
   (v (version-from-scm $mproject) "cache") => anything
   (provided
-    (spit "/X/version.clj" (as-checker (partial re-find #"1.2.3-1-0x[0-9a-f]{4,}"))) => ..result..))
+    (spit #"/X/version.(clj|edn)" (as-checker (partial re-find #"1.2.3-1-0x[0-9a-f]{4,}"))) => ..result..))
```
Comment (cch1) [inline test/integration/leiningen/v.clj]: This should test for both, not either

File: project.clj (hunk 2 above)
Comment (cch1) [reply project.clj]: This issue was solved a while ago on master.

**Comment by jgrodziski** [comment]
Hi Chris, ok, thanks for your review and sorry for the late reply. Will finish this PR now.
Actually, I would like to use the version file in Clojurescript projects. 
Are you ok adding a cljs or cljc version as well ?

**Comment by cch1** [comment]
I've got no problem with adding a `cljs` or `cljc` or even `cljx` option.  A simple `case` statement is all that is needed.

**Comment by jgrodziski** [comment]
Sorry Chris, I made a branch in my fork and another PR #19 instead of pushing onto master to update that one...Tell me the one you want to use and the one to close.
